### PR TITLE
16 create an API route for deleting a volunteer

### PR DESCRIPTION
**Pull request by ethanhmaness**
# Description
Added API route and server func for deleting volunteer from database by ID. Note that the {strict: false} option is required for it to work on objects that do not have a pre-existing softDelete field set to false.

## Relevant issue(s)

#16 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have assigned reviewers to this PR

### CODE REVIEW

File: src/server/actions/Volunteers.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+ */
+export async function softDeleteVolunteer(
+  volunteerId: string
+): Promise<VolunteerEntity | null> {
+  await dbConnect();
+
+  const res: VolunteerEntity | null = await VolunteerSchema.findByIdAndUpdate(
+    volunteerId,
+    { softDelete: true },
+    { strict: false } //Makes sure to add a softDelete field to the volunteer if it does not already have one
```
Comment (andlrutt) [inline src/server/actions/Volunteers.ts:18]: Looks like we had a miss when setting up the model for this. Can you add it to the `VolunteerSchema` model and have it be required and default to false? That should remove the need for this line

File: src/app/api/volunteers/[volunteerId]/route.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import { softDeleteVolunteer } from '@/server/actions/Volunteers';
+import { zObjectId } from '@/types/dataModel/base';
+import { NextRequest, NextResponse } from 'next/server';
+
+// @route DELETE /api/volunteers/[organizationId] - Soft deletes an organization
```
Comment (andlrutt) [inline src/app/api/volunteers/[volunteerId]/route.ts:5]: soft deletes a volunteer?

**Review by andlrutt — CHANGES_REQUESTED** [review]

File: src/server/actions/Volunteers.ts [hunk 3]
```diff
@@ -1,15 +1,33 @@
...
+ * Soft delete a volunteer
+ * @param volunteerId The Id of the volunteer to be deleted
+ * @returns The volunteer in the database before soft-deletion, or null
+ */
+export async function softDeleteVolunteer(
+  volunteerId: string
+): Promise<VolunteerEntity | null> {
+  await dbConnect();
+
+  const res: VolunteerEntity | null = await VolunteerSchema.findByIdAndUpdate(
```
Comment (zaviermiller) [inline src/server/actions/Volunteers.ts:18]: should this be `VolunteerResponse`?

Comment (zaviermiller) [reply src/server/actions/Volunteers.ts:18]: @andlrutt

Comment (andlrutt) [reply src/server/actions/Volunteers.ts:18]: yes. thought I commented on this, may have forgotten to submit the review 😁

**Review by zaviermiller — APPROVED** [review]
lgtm

**Pull request merged**